python/cudf/cudf/_lib/pylibcudf/hashing.pyx

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+from libc.stdint cimport uint32_t, uint64_t
+from libcpp.memory cimport unique_ptr
+from libcpp.utility cimport move
+
+from cudf._lib.cpp.column.column cimport column
+from cudf._lib.cpp.hash cimport (
+    md5 as cpp_md5,
+    murmurhash3_x64_128 as cpp_murmurhash3_x64_128,
+    murmurhash3_x86_32 as cpp_murmurhash3_x86_32,
+    sha1 as cpp_sha1,
+    sha224 as cpp_sha224,
+    sha256 as cpp_sha256,
+    sha384 as cpp_sha384,
+    sha512 as cpp_sha512,
+    xxhash_64 as cpp_xxhash_64,
+)
+from cudf._lib.cpp.table.table cimport table
+
+from .column cimport Column
+from .table cimport Table
+
+
+cpdef Column hash(Table input, object method, object seed):
```

Comment:
Please implement this dispatching logic in cudf -- not pylibcudf. We are getting rid of the top level `hash` function in libcudf and will only expose algorithm-specific functions, so pylibcudf should do the same.

Eventually I want to deprecate and remove the `hash` function from cudf Cython and the `hash_values` function from cudf Python in favor of only exposing specific algorithms via pylibcudf.